src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class RightTriangle extends Figure {
+    public static final int MAXIMUM_SIZE = 50;
+    private int firstLeg;
+    private int secondLeg;
+    private Random random = new Random();
+
+    @Override
+    public double getArea() {
+        return 0.5 * firstLeg * secondLeg;
+    }
+
+    @Override
+    public void randomizeProperty() {
+        firstLeg = random.nextInt(MAXIMUM_SIZE);
+        secondLeg = random.nextInt(MAXIMUM_SIZE);
```

Comment:
The method 'randomizeProperty' should not exist in the RightTriangle class. Instead, the random values should be generated in the FigureSupplier and passed to the constructor of RightTriangle. Please remove this method.